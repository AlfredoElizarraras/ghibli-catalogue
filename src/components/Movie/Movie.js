import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import movieImage from '../../utils/data';
import styles from './movie.module.css';

const Movie = ({ movie }) => (
  <>
    <nav className={styles.navbar}>
      <Link to="/">
        <img className={styles.arrow} src="https://cdn.iconscout.com/icon/free/png-512/back-arrow-1767531-1502435.png" alt="back to movies" />
      </Link>

      <h1>{ movie.title }</h1>
    </nav>

    <div className={styles.movie}>
      <div className={styles.imageContainer}>
        <img className={styles.movieImage} alt={movie.title} src={movieImage[movie.id]} />
      </div>
      <div className={styles.movieInfo}>
        <div className={styles.movieInfoHighLights}>
          <div>
            <div>
              <div>Release year</div>
              { movie.release_date }
            </div>
            <div className={styles.separator} />
            <div>
              <div>Score:</div>
              { movie.rt_score }
              / 100
            </div>
          </div>
          <div>
            <div>
              <div>Director:</div>
              { movie.director }
            </div>
            <div className={styles.separator} />
            <div>
              <div>Producer:</div>
              { movie.producer }
            </div>
          </div>
        </div>
        <div>{ movie.description }</div>
      </div>
    </div>
  </>
);

Movie.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    producer: PropTypes.string.isRequired,
    release_date: PropTypes.string.isRequired,
    rt_score: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  }).isRequired,
};

export default Movie;
