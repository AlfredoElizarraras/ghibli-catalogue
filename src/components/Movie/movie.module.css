.movie {
  display: flex;
  flex-direction: column;
  margin: 1rem;
}

.movieInfo {
  display: flex;
  flex-direction: column;
  padding-left: 2rem;
  padding-right: 2rem;
  margin-bottom: 2rem;
}

.movieInfoHighLights {
  display: flex;
  justify-content: space-between;
  margin-bottom: 1rem;
}

.navbar {
  display: flex;
  align-items: center;
  padding-left: 1rem;
  padding-right: 2rem;
  height: 50px;
  background-color: #90caf9;
  font-size: 1.5rem;
}

.imageContainer {
  display: flex;
  justify-content: center;
}

.movieImage {
  height: 500px;
  width: 400px;
}

.arrow {
  margin-right: 2rem;
  height: 30px;
  width: 30px;
}

.separator {
  margin: 1rem 0;
}

@media only screen and (min-width: 600px) {
  .movie {
    flex-direction: row;
  }

  .imageContainer {
    width: 50%;
  }
}
