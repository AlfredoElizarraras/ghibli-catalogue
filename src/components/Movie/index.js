import { useParams } from 'react-router-dom';
import { connect } from 'react-redux';
import { getMovie } from '../../redux/selectors/selectors';
import Movie from './Movie';

const mapStateToProps = state => {
  const { movieId } = useParams();

  return { movie: getMovie(state, movieId)[0] };
};

export default connect(mapStateToProps, null)(Movie);
