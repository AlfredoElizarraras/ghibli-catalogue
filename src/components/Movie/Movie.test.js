import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Movie from './Movie';

describe('Movie', () => {
  test('should render without crashing', () => {
    const props = {
      movie: {
        id: '1',
        title: 'Movie title',
        description: 'Movie description',
        director: 'Movie director',
        producer: 'Movie producer',
        release_date: 'Movie release date',
        rt_score: 'Movie rate score',
        url: 'Movie URL',
      },
    };

    const tree = renderer.create(
      <BrowserRouter>
        <Movie movie={props.movie} />
      </BrowserRouter>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
