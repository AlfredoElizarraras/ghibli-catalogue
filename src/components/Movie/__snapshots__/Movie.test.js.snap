// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should render without crashing 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="back to movies"
        className="arrow"
        src="https://cdn.iconscout.com/icon/free/png-512/back-arrow-1767531-1502435.png"
      />
    </a>
    <h1>
      Movie title
    </h1>
  </nav>,
  <div
    className="movie"
  >
    <div
      className="imageContainer"
    >
      <img
        alt="Movie title"
        className="movieImage"
      />
    </div>
    <div
      className="movieInfo"
    >
      <div
        className="movieInfoHighLights"
      >
        <div>
          <div>
            <div>
              Release year
            </div>
            Movie release date
          </div>
          <div
            className="separator"
          />
          <div>
            <div>
              Score:
            </div>
            Movie rate score
            / 100
          </div>
        </div>
        <div>
          <div>
            <div>
              Director:
            </div>
            Movie director
          </div>
          <div
            className="separator"
          />
          <div>
            <div>
              Producer:
            </div>
            Movie producer
          </div>
        </div>
      </div>
      <div>
        Movie description
      </div>
    </div>
  </div>,
]
`;
